lizrd/scripts/grid.py

```diff
@@ -26,90 +27,88 @@
 )
 from lizrd.support.code_versioning_support import copy_and_version_code
 
-RUNNER = "research.reinitialization.train.reinit_train"
-
-
-# ^ - grid over that
-# * - apply function
-PARAMS = {
-    "project_name": f"{os.getenv('USER')}/mp",
-    "name": "mp",
-    "ff_layer": "regular",
-    "batch_size": 128,
-    "cutoff": 128,
-    "^mixed_precision": [True, False],
-    "tags": ["test"],
-    "use_clearml": True,
-    "pruner_n_steps": 100,
-}
-
-TIME = "1-00:00:00"
-GRES = "gpu:titanv:1"
-DRY_RUN = False
-CODE_PATH = os.getcwd()
-INTERACTIVE_DEBUG = False
-RUNS_MULTIPLIER = 1
-PUSH_TO_GIT = False
-COPIED_CODE_PATH = ""
 if __name__ == "__main__":
     runner = get_machine_backend()
-    HF_DATASETS_CACHE = get_cache_path(runner)
-    SINGULARITY_IMAGE = get_sparsity_image(runner)
-    NODELIST = None
-    N_GPUS = 1
-    CPUS_PER_GPU = 8
-    CUDA_VISIBLE_DEVICES = None
     PROCESS_CALL_FUNCTION = lambda args, env: subprocess.run(
         [str(arg) for arg in args if arg is not None], env=env
     )
-
     try:
         path = sys.argv[1]
     except IndexError:
-        raise ValueError("config path must be specified as the first argument")
+        raise ValueError("No config path specified. Aborting...")
 
     if path.endswith(".yaml"):
-        grid_args = yaml.safe_load(open(sys.argv[1]))
+        configs = yaml.safe_load_all(open(sys.argv[1]))
```

Comment:
what, does it work? how does it recognize end/beginning of yaml config? can you provide an example file with 2 or more configs?

Reply:
see the diff now. Got tests! The engineering gods are smiling upon us